lib/pure/net.nim

```diff
@@ -755,21 +755,20 @@ proc bindAddr*(socket: Socket, port = Port(0), address = "") {.
   ## Binds ``address``:``port`` to the socket.
   ##
   ## If ``address`` is "" then ADDR_ANY will be bound.
+  var realaddr = address
+  if realaddr == "":
+    case socket.domain
+    of AF_INET6: realaddr = "::"
```

Comment:
It kinda irks me that these strings will be created, just to be thrown away in the end. Ah well...

Reply:
realaddr is being used... do you have any alternative?

Reply:
@Varriount with --gc:destructors these are pretty much zero cost.